src/axom/quest/IntersectionShaper.hpp

```diff
@@ -2906,6 +2990,38 @@ class IntersectionShaper : public Shaper
     return isTri;
   }
 
+  /*!
+   * \brief Returns the dimension of the computational mesh
+   * \return dim The dimensions of the mesh (expected is 2 or 3, -1 in case of failure)
+   */
+  int getCompMeshDim()
+  {
+    int dim = -1;
+  #if defined(AXOM_USE_MFEM)
+    if(m_dc != nullptr)
+    {
+      dim = this->getDC()->GetMesh()->SpaceDimension();
+    }
+  #endif
+  #if defined(AXOM_USE_CONDUIT)
+    if(m_bpGrp != nullptr)
+    {
+      std::string mesh_type = m_bpGrp->getView("topologies/mesh/elements/shape")->getString();
+      if(mesh_type == "hex")
+      {
+        dim = 3;
+      }
+      else if(mesh_type == "quad")
+      {
+        dim = 2;
+      }
+    }
+  #endif
+
+    SLIC_ERROR_IF(!(dim == 2 || dim == 3), "Invalid computational mesh dimension");
+    return dim;
+  }
```

Comment:
Looks like this is keying of the computational mesh, rather than from Klee.

In https://github.com/LLNL/axom/issues/1564 I'm proposing a better way to control the shape dimension from Klee. 
Not sure if that will help here...

Reply:
> Looks like this is keying of the computational mesh, rather than from Klee.

Yes, it's doing exactly that, since the dimension in the klee `yaml` file is currently always `dimensions: 2` for c2c contours.

> In https://github.com/LLNL/axom/issues/1564 I'm proposing a better way to control the shape dimension from Klee.
Not sure if that will help here...

I think this will help a case like this. Driving the 2D or 3D option from the computation mesh instead of the shapes specification feels weird. Open to adopting a better way to do things.